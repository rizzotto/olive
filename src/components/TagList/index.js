import React, { useState } from 'react';
import useStyles from './styles';
import PropTypes from 'prop-types';
import Tag from '../Tag';

function TagList({ onClick, tags, horizontal, multiSelect, ...others }) {
  const classes = useStyles({ horizontal });
  const [tagsArray, setTagsArray] = useState(tags);

  function handleOnClick(id) {
    const newTags = tagsArray.map(tag => {
      if (tag.tag_id === id) return { ...tag, selected: !tag.selected };

      return multiSelect ? tag : { ...tag, selected: false };
    });

    setTagsArray(newTags);

    if (onClick) onClick(newTags);
  }

  return (
    <div className={classes.root}>
      {tagsArray.map(tag => (
        <Tag
          key={tag.tag_id}
          customStyle={{ margin: '5px 3px' }}
          selected={tag.selected}
          name={tag.title}
          onClick={() => handleOnClick(tag.tag_id)}
          {...others}
        />
      ))}
    </div>
  );
}

TagList.propTypes = {
  onClick: PropTypes.func,
  tags: PropTypes.array,
  horizontal: PropTypes.bool,
  multiSelect: PropTypes.bool,
};

TagList.defaultProps = {
  onClick: () => {},
  tags: [],
  horizontal: true,
  multiSelect: true,
};

export default TagList;
