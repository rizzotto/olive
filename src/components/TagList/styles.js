import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  root: {
    display: 'flex',
    margin: '10px -20px',
    padding: '0 20px',

    justifyContent: props => !props.horizontal && 'flex-start',
    flexWrap: props => !props.horizontal && 'wrap',
    overflowX: props => props.horizontal && 'auto',
    '&::-webkit-scrollbar': {
      display: 'none',
    },
  },
}));

export default useStyles;
