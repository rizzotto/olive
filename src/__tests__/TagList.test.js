import React from 'react';
import renderer from 'react-test-renderer';
import TagList from '../components/TagList';

test('renders TagList', () => {
  const tags = [
    { tag_id: 1, title: 'test1', selected: false },
    { tag_id: 2, title: 'test2', selected: false },
    { tag_id: 3, title: 'test3', selected: false },
    { tag_id: 4, title: 'test4', selected: true },
  ];

  const tree = renderer.create(<TagList tags={tags} />).toJSON();
  expect(tree).toMatchSnapshot();
});
