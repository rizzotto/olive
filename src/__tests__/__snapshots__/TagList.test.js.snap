// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TagList 1`] = `
<div
  className="makeStyles-root-1 makeStyles-root-2"
>
  <div
    className="makeStyles-root-3 makeStyles-root-4"
    style={
      Object {
        "margin": "5px 3px",
      }
    }
  >
    test1
  </div>
  <div
    className="makeStyles-root-3 makeStyles-root-5"
    style={
      Object {
        "margin": "5px 3px",
      }
    }
  >
    test2
  </div>
  <div
    className="makeStyles-root-3 makeStyles-root-6"
    style={
      Object {
        "margin": "5px 3px",
      }
    }
  >
    test3
  </div>
  <div
    className="makeStyles-root-3 makeStyles-root-7"
    style={
      Object {
        "margin": "5px 3px",
      }
    }
  >
    test4
  </div>
</div>
`;
